src/group_theory/subgroup.lean

```diff
@@ -155,6 +155,10 @@ lemma coe_to_submonoid (K : subgroup G) : (K.to_submonoid : set G) = K := rfl
 instance (K : subgroup G) [d : decidable_pred (∈ K)] [fintype G] : fintype K :=
 show fintype {g : G // g ∈ K}, from infer_instance
 
+@[to_additive]
+noncomputable instance subgroup.fintype_noncomputable (K : subgroup G) [fintype G] : fintype K :=
+fintype.of_injective coe subtype.coe_injective
```

Comment:
This seems like a bad instance to have, as it can easily result in diamonds. Why isn't the instance on line 155 sufficient here?

Reply:
It seems like lean wasn't able to figure out the `[d : decidable_pred (∈ K)]` hypothesis. What's the best way to deal with this?

Reply:
Introduce that hypotheses with `haveI := classical.dec_pred (∈ K)` in the place you need it?

Reply:
The problem is that `fintype.card H` is used in the statement of the lemma `subgroup_eq_top_of_mem_swap`.

Reply:
Where specifically was it you needed it?